proxy/pkg/zdmproxy/clusterconn.go

```diff
@@ -257,7 +266,15 @@ func (cc *ClusterConnector) runResponseListeningLoop() {
 		for {
 			response, err := readRawFrame(bufferedReader, connectionAddr, cc.clusterConnContext)
 			protocolErrResponseFrame, err, errCode := checkProtocolError(response, cc.ccProtoVer, err, protocolErrOccurred, string(cc.connectorType))
-
+			if response != nil {
+				if _, ok := cc.internalRequests[response.Header.StreamId]; ok {
```

Comment:
I'm a bit confused here, the response stream id for "internal requests" will always be -1 here so you can't have more than 1 internal request at the same time.

If you change the stream id so this is the stream id that is assigned then you will have the problem of not knowing whether a response is for an internal request or not.

Do we even need to track these internal requests? Can't we just check that the stream id of the frame is -1 after the call to `ReleaseId()` in `clusterconn.runResponseListeningLoop()` and if it is then we just don't send it to the response channel? In this case we could just delete `internalRequests` map